lib/galaxy/config/__init__.py

```diff
@@ -313,6 +314,13 @@ def root_join(val):
                 setattr(self, var + '_set', False)
             setattr(self, var, [root_join(x) for x in paths])
 
+        # update kwargs to set unset renamed options
```

Comment:
The problem is that we use the `kwargs` arguments passed to `GalaxyAppConfiguration` very early: [here](https://github.com/galaxyproject/galaxy/blob/5b9b1d26373040c54b14b7b49855a319f08a9f9c/lib/galaxy/config/__init__.py#L121) and even [here](https://github.com/galaxyproject/galaxy/blob/5b9b1d26373040c54b14b7b49855a319f08a9f9c/lib/galaxy/config/__init__.py#L116). There is (or there may be) code that depends on the assumption that these arguments will not play any further role in loading (or [reloading](https://github.com/galaxyproject/galaxy/blob/5b9b1d26373040c54b14b7b49855a319f08a9f9c/lib/galaxy/config/__init__.py#L874)) the config, once they have been processed in `_update_raw_config_from_kwargs()`.

I agree that seamless backward compatibility is a requirement here. So, if we have to modify these arguments, let's do it before any subsequent processing, and, maybe, let's be very explicit about this:
```python
def __init__(self, **kwargs):
    self._append_renamed_options(kwargs)  # or any other method name
    self.config_dict = kwargs   # and the rest stays as is
```
What do you think?

Reply:
Makes sense, done in the latest commit, thanks!